immutables/map.py

```diff
@@ -431,7 +432,12 @@ def __iter__(self):
         return iter(self.__root.items())
 
 
-class Map:
+K = TypeVar('K', bound=Hashable)
+V = TypeVar('V', bound=Any)
+D = TypeVar('D', bound=Any)
+
+
+class Map(Mapping[K, V]):
```

Comment:
Do we need this if we add a pyi file anyways?

Reply:
It's not entirely clear to me in what circumstances this implementation is used as opposed to the C one, but if this ever is used it should probably inherit from Mapping so that `Map[foo, bar]` doesn't raise a TypeError. Note that I am not experienced with writing C extensions and so this was mostly a hack from my point of view. If you feel like things look weird, they probably are but I am happy for the feedback and can put some time into getting it right :)

Reply:
We can't inherit it from Mapping, as we can only do that for the pure Python version.  So both shouldn't be inherited from Mapping.

I'm working on a PEP to have `immutables.Map` in the stdlib.  If we do that we can just call `Mapping.register(Map)` and be done with this.

Reply:
Oh right, so should we make the pure Python version also implement `__class_get_item__` instead? Because I believe `Mapping.register(Map)` is not sufficient for making `Map[foo, bar]` return `Map`?

Very cool with a PEP and stdlib support! :)